core/model/modx/transport/modtransportpackage.class.php

```diff
@@ -661,33 +722,45 @@ protected function _getByFsockopen($url) {
         $purl = parse_url($url);
         $host = $purl['host'];
         $path = !empty($purl['path']) ? $purl['path'] : '/';
-        if (!empty($purl['query'])) { $path .= '?'.$purl['query']; }
+        if (!empty($purl['query'])) {
+            $path .= '?' . $purl['query'];
+        }
         $port = !empty($purl['port']) ? $purl['port'] : '80';
 
         $timeout = 10;
         $response = '';
         $fp = @fsockopen($host,$port,$errno,$errstr,$timeout);
 
         if( !$fp ) {
-            $this->xpdo->log(xPDO::LOG_LEVEL_ERROR,'Could not retrieve from '.$url);
+            $this->xpdo->log(xPDO::LOG_LEVEL_ERROR, 'Could not retrieve from ' . $url);
         } else {
-            fwrite($fp, "GET $path ".$_SERVER['SERVER_PROTOCOL']."\r\n" .
-                "Host: $host\r\n" .
-                "User-Agent: Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.0.3) Gecko/20060426 Firefox/1.5.0.3\r\n" .
-                "Accept: */*\r\n" .
-                "Accept-Language: en-us,en;q=0.5\r\n" .
-                "Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\n" .
-                "Keep-Alive: 300\r\n" .
-                "Connection: keep-alive\r\n" .
-                "Referer: http://$host\r\n\r\n");
-
-          while ($line = fread($fp, 4096)) {
-             $response .= $line;
-          }
-          fclose($fp);
-
-          $pos = strpos($response, "\r\n\r\n");
-          $response = substr($response, $pos + 4);
+            $out = implode("\r\n", array(
+                "GET $path " . $_SERVER['SERVER_PROTOCOL'],
+                "Host: $host",
+                "User-Agent: Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.0.3) Gecko/20060426 Firefox/1.5.0.3",
+                "Accept: */*",
+                "Accept-Language: en-us,en;q=0.5",
+                "Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7",
+                "Keep-Alive: 300",
+                "Connection: keep-alive",
+                "Referer: http://$host",
```

Comment:
I don't think it has any influence on how the code works but just to be consistent it would be nice if the referer would use the same protocol (http or https) as the "host".